### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by Vayts**

### CODE REVIEW

**Review by polinavafik — APPROVED** [review]
very sophisticated 🎀

File: src/convertToObject.js [hunk 1]
```diff
@@ -3,16 +3,21 @@
...
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString.split(';')
+    .filter((item) => item.trim() !== '')
```
Comment (BudnikOleksii) [inline src/convertToObject.js:13]: [CODE STYLE] - All chained methods called on the same level should have same indentation. This makes code much more readable.
image
Also, in the filter you don't need to compare with an empty string you can just return `item.trim()` and all falsy values will be excluded. `.filter((item) => item.trim())`

File: src/convertToObject.js [hunk 2]
```diff
@@ -3,16 +3,21 @@
...
+    .map((item) => item.split(':'))
+    .reduce((acc, item) => {
+      return {
+        ...acc,
+        [item[0].trim()]: item[1].trim(),
+      };
```
Comment (BudnikOleksii) [inline src/convertToObject.js:19]:
```suggestion
    .reduce((acc, [key, value]) => {
      return {
        ...acc,
        [key.trim()]: value.trim(),
      };
```
You can use destructuring.

File: src/convertToObject.js [hunk 3]
```diff
@@ -3,16 +3,22 @@
...
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .filter((item) => item.trim() !== '')
+    .map((item) => item.split(':'))
+    .reduce((acc, [key, value]) => {
```
Comment (Moroz-Dmytro) [inline src/convertToObject.js:16]: We can combine these 3 loops to just 1 `reduce`

File: src/convertToObject.js [hunk 4]
```diff
@@ -3,16 +3,22 @@
...
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
```
Comment (Moroz-Dmytro) [inline src/convertToObject.js:13]: you can create variables for these magic values ';' ':'

**Review by Moroz-Dmytro — CHANGES_REQUESTED** [review]
Good work 👍 
But let's improve your solution

File: src/convertToObject.js [hunk 5]
```diff
@@ -3,16 +3,32 @@
...
+    .split(SEMICOLON)
+    .reduce((acc, item) => {
+      const [key, value] = item.split(COLON);
+
+      if (key.trim() !== '') {
+        return {
+          ...acc,
+          [key.trim()]: value.trim(),
+        };
+      }
```
Comment (Moroz-Dmytro) [inline src/convertToObject.js:28]:
```suggestion

      if (key.trim()) {
        acc[key.trim()] = value.trim();
      }
```

**Review by Moroz-Dmytro — APPROVED** [review]
GJ 👍 
But check comment below

**Pull request closed without merging**